source/tool/tests/software/mock_helper.py

```diff
@@ -130,6 +130,55 @@ def read_phys_mem(self, pa_hi, pa_lo, length):
         else:
             return "\xFF" * length
 
+class DSDTParsingHelper(ACPIHelper):
+    """Test helper containing generic descriptor for RSDP, RSDT, and FADT to parse DSDT
+
+    Default entry for DSDT and X_DSDT inside FADT is 0x0
+
+    One additional region is defined:
+      * FADT table [0x400, 0x514]
+    """
+    USE_FADT_WITH_X_DSDT = True
+
+    FADT_ADDRESS = 0x400
+    DSDT_ADDRESS = 0x0
+    X_DSDT_ADDRESS = 0x0
+
+    def _ensure_fadt_in_sdt(self, SDT_ENTRIES):
+        if self.FADT_ADDRESS not in SDT_ENTRIES:
+            SDT_ENTRIES.append(self.FADT_ADDRESS)
```

Comment:
...Not sure. `_ensure_fadt_in_sadt` was written so that classes that derive from `DSDTParsingHelper` need not to worry about adding FADT entry to R/XSDT.

On a second thought, expecting the user to add an FADT entry to R/XSDT sounds better, because the user knows that they are changing a constant and that the constant is not going to be changed by anything else (therefore it's their responsibility to fill in the correct entries to R/XSDT).

By moving it to instance attribute, do you mean the derived classes need to add their needed entries to R/XSDT in their own `__init__` method?